elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -0,0 +1,151 @@
+        splitFilters();
+    }
+
+    /**
+     * @return {@link Query} query object with all the parameters provided by user.
+     */
+    public Query getQuery() {
+        return Query.builder()
+                .schema(schema)
+                .metrics(metricMap)
+                .groupDimensions(dimensions)
+                .timeDimensions(timeDimensions)
+                .whereFilter(whereFilter)
+                .havingFilter(havingFilter)
+                .sorting(entityProjection.getSorting())
+                .pagination(entityProjection.getPagination())
+                .build();
+    }
+
+    /**
+     * Gets whereFilter and havingFilter based on provided filter expression from {@link EntityProjection}.
+     */
+    private void splitFilters() {
+        FilterExpression filterExpression = entityProjection.getFilterExpression();
+        if (filterExpression == null) {
+            whereFilter = null;
+            havingFilter = null;
+            return;
+        }
+        SplitFilterExpressionVisitor visitor = new SplitFilterExpressionVisitor(schema);
+        FilterConstraints constraints = filterExpression.accept(visitor);
+        whereFilter = constraints.getWhereExpression();
+        havingFilter = constraints.getHavingExpression();
+    }
+
+    /**
+     * Gets dimensions and timeDimensions based on relationships and attributes from {@link EntityProjection}.
+     * @param dimensions Empty set of {@link Dimension} objects.
+     * @param timeDimensions Empty set of {@link TimeDimension} objects.
+     */
+    public void resolveDimensionLists(Set<Dimension> dimensions, Set<TimeDimension> timeDimensions) {
+        Set<String> dimensionNames = getRelationships();
+        Set<String> metricNames = getAttributes(); // time dimensions are under attribute in entity projection
+        for (Dimension dimension : schema.getDimensions()) {
+            if (dimension instanceof TimeDimension && metricNames.contains(dimension.getName())) {
+                timeDimensions.add((TimeDimension) dimension);
+            }
+            else if (dimensionNames.contains(dimension.getName()) || metricNames.contains(dimension.getName())) {
+                dimensions.add(dimension);
+            }
+        }
+    }
+
+    /**
+     * Gets metrics based on attributes from {@link EntityProjection}.
+     * @param metrics Empty set of {@link Metric} objects.
+     */
+    public void resolveMetricList(Set<Metric> metrics) {
+        Set<String> metricNames = getAttributes();
+        for (Metric metric : schema.getMetrics()) {
+            if (metricNames.contains(metric.getName())) {
+                metrics.add(metric);
+            }
+        }
+    }
+
+    /**
+     * Constructs map between {@link Metric} objects and the type of {@link Aggregation} we want to use.
+     * @param metrics Set of {@link Metric} objects.
+     */
+    public void resolveMetricMap(Set<Metric> metrics) {
+        metricMap = new LinkedHashMap<>();
```

Comment:
Maybe it would be better if we put the instantiation of `metricMap` in the constructor.